catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/report/RegistryReportBuilder.java

```diff
@@ -0,0 +1,150 @@
+            String handlebarTemplate) throws IOException {
+
+        Map<String, Object> reportMap = getSummaryMap(metacard);
+        Handlebars handlebars = getHandlebarsFromTemplate(templateLoader);
+        Template template = handlebars.compile(handlebarTemplate);
+        String html = template.apply(reportMap);
+        html = html.replaceAll("\n", "");
+
+        return html;
+    }
+
+    public String getHTMLFromRegistryPackage(RegistryPackageType registryPackage,
+            ClassPathTemplateLoader templateLoader, String handlebarTemplate) throws IOException {
+
+        Map<String, Object> reportMap = new HashMap<>();
+        if (handlebarTemplate.equals("organizations")) {
+            reportMap = getOrganizationMap(registryPackage);
+        } else if (handlebarTemplate.equals("report")) {
+            reportMap = getFullRegistryMap(registryPackage);
+        }
+        Handlebars handlebars = getHandlebarsFromTemplate(templateLoader);
+        Template template = handlebars.compile(handlebarTemplate);
+        String html = template.apply(reportMap);
+
+        return html;
+    }
+
+    private Handlebars getHandlebarsFromTemplate(ClassPathTemplateLoader templateLoader) {
+        Handlebars handlebars = new Handlebars(templateLoader);
+        handlebars.registerHelpers(new HandlebarsHelper());
+        handlebars.registerHelpers(StringHelpers.class);
+
+        return handlebars;
+    }
+
+    private Map<String, Object> getSummaryMap(Metacard metacard) {
+        Map<String, Object> summaryMap = new HashMap<>();
+
+        try {
+            String pointOfContact = RegistryUtility.getStringAttribute(metacard,
+                    Metacard.POINT_OF_CONTACT,
+                    null);
+            String organizationName = RegistryUtility.getStringAttribute(metacard,
+                    RegistryObjectMetacardType.ORGANIZATION_NAME,
+                    null);
+            String organizationAddress = RegistryUtility.getStringAttribute(metacard,
+                    RegistryObjectMetacardType.ORGANIZATION_ADDRESS,
+                    null);
+            String organizationEmail = RegistryUtility.getStringAttribute(metacard,
+                    RegistryObjectMetacardType.ORGANIZATION_EMAIL,
+                    null);
+            String organizationPhoneNumber = RegistryUtility.getStringAttribute(metacard,
+                    RegistryObjectMetacardType.ORGANIZATION_PHONE_NUMBER,
+                    null);
+
+            summaryMap.put("Point of Contact", pointOfContact);
+            summaryMap.put("Organization Name", organizationName);
+            summaryMap.put("Address", organizationAddress);
+            summaryMap.put("Email Address", organizationEmail);
+            summaryMap.put("Phone Number", organizationPhoneNumber);
+        } catch (NullPointerException e) {
+            LOGGER.debug("Null value not put into summary map" + e.toString());
+        }
```

Comment:
Maybe there is another way we can handle the missing values here, perhaps by making the default value "not provided" or something. Then you would avoid NPE and partially formed summaryMaps altogether.

Too bad you couldn't just use the constants as the labels also... then you could just stream a list of desired fields instead.